akka-remote/src/main/scala/akka/remote/PhiAccrualFailureDetector.scala

```diff
@@ -122,7 +120,11 @@ class PhiAccrualFailureDetector(
         // this is a known connection
         val interval = timestamp - latestTimestamp
         // don't use the first heartbeat after failure for the history, since a long pause will skew the stats
-        if (isAvailable(timestamp)) oldState.history :+ interval
+        if (isAvailable(timestamp)) {
+          if (interval >= (acceptableHeartbeatPauseMillis / 2))
+            logger.warning(s"heartbeat interval is growing too large: $interval millis")
```

Comment:
logging should be done with the Akka `EventStream`:
```
eventStream.publish(Warning(getClass.getName, getClass, s"heartbeat interval is growing too large: $interval millis"))
```

The `eventStream` is passed the constructor `def this(config: Config, ev: EventStream)` but unfortunately ignored there. We have to do some tricks to keep it backwards compatible. Add another constructor:
```
 class PhiAccrualFailureDetector(
   val threshold:                Double,
   val maxSampleSize:            Int,
   val minStdDeviation:          FiniteDuration,
   val acceptableHeartbeatPause: FiniteDuration,
   val firstHeartbeatEstimate:   FiniteDuration,
   eventStream: EventStream)(
   implicit
   clock: Clock) extends FailureDetector {
```

The old constructor must be added for backwards compatibility and pass `null` as `eventStream`, and the logging must be done with `if (eventStream != null) eventStream.publish...`

Somewhat messy, but that's needed for backwards compatibility.